SU2_CFD/src/solvers/CHeatSolver.cpp

```diff
@@ -69,6 +65,27 @@ CHeatSolver::CHeatSolver(CGeometry *geometry, CConfig *config, unsigned short iM
   nMarker = config->GetnMarker_All();
 
   CurrentMesh = iMesh;
+
+  /*--- Store the number of vertices on each marker for deallocation later ---*/
+
+  nVertex.resize(nMarker);
+  for (iMarker = 0; iMarker < nMarker; iMarker++)
+    nVertex[iMarker] = geometry->nVertex[iMarker];
+
+  /*--- Allocates a 2D array with variable "outer" sizes and init to 0. ---*/
+
+  auto Alloc2D = [](unsigned long M, const vector<unsigned long>& N, vector<vector<su2double> >& X) {
+    X.resize(M);
+    for(unsigned long i = 0; i < M; ++i) X[i].resize(N[i], 0.0);
+  };
+
+  /*--- Allocates a 3D array with variable "middle" sizes and init to 0. ---*/
+
+  auto Alloc3D = [](unsigned long M, const vector<unsigned long>& N, unsigned long P, vector<su2activematrix>& X) {
+    X.resize(M);
+    for(unsigned long i = 0; i < M; ++i) X[i].resize(N[i],P) = su2double(0.0);
+  };
```

Comment:
This is taken from the `CEulerSolver` constructor. Maybe `AllocND` could be defined in `CSolver.hpp` already?

Reply:
If you can avoid code duplication, I am in for moving that to CSolver 👍

Reply:
Free function in container helpers or something

Reply:
done in commit 3f65e8e